src/routes/app/+page.svelte

```diff
@@ -116,15 +102,29 @@
         <input type="checkbox" name="select_all" />
         Select all
       </label>
-      <Paginator
-        slot="center"
-        {page}
-        totalPages={total_pages}
-        has_next={Boolean(next)}
-        has_previous={Boolean(previous)}
-        on:next={load_next}
-        on:previous={load_previous}
-      />
+
+      <svelte:fragment slot="center">
+        {#await searchResults then sr}
+          {@const count = sr.count}
+          {@const total_pages = Math.ceil(count / per_page)}
+          {@const next = sr.next}
+          {@const previous = sr.previous}
+          <Paginator
+            {page}
+            totalPages={total_pages}
+            has_next={Boolean(next)}
+            has_previous={Boolean(previous)}
+            on:next={(e) => {
+              page = Math.min(total_pages, page + 1);
+              load(next);
+            }}
+            on:previous={(e) => {
+              page = Math.max(0, page - 1);
```

Comment:
```suggestion
              page = Math.max(1, page - 1);
```
If page is 0, the pagination loses its "Page M of N" text